packages/tailwindcss/src/theme.ts

```diff
@@ -227,6 +228,7 @@ export type ThemeKey =
   | '--scroll-padding'
   | '--sepia'
   | '--shadow'
+  | `--shadow-${string}`
```

Comment:
I think a better way of doing this would be to update the `get` function instead:

```diff
-  get(themeKeys: ThemeKey[]): string | null {
+  get(themeKeys: (ThemeKey | `${ThemeKey}-${string}`)[]): string | null {
```

That way you get this for free for every property, but only when using `theme.get`